source/translated_table_model.ts

```diff
@@ -157,13 +157,15 @@ export class TranslatedTableModel extends TableModel {
         return rowIndex;
       }
     })();
+    const trans = this.translation.slice();
```

Comment:
While this will fix the bug, it will be too costly of an operation unfortunately. Consider only making a copies of the values you actually need at the time that you need them.